SignalServiceKit/src/Loki/API/LokiFileServerProxy.swift

```diff
@@ -0,0 +1,106 @@
+                "body" : String(bytes: parametersAsData, encoding: .utf8),
+                "endpoint": endpoint,
+                "method" : request.httpMethod,
+                "headers" : headers
+            ]
+            let proxyRequestParametersAsData = try JSONSerialization.data(withJSONObject: proxyRequestParameters, options: [])
+            let ivAndCipherText = try DiffieHellman.encrypt(proxyRequestParametersAsData, using: symmetricKey)
+            let proxyRequestHeaders = [
+                "X-Loki-File-Server-Target" : "/loki/v1/secure_rpc",
+                "X-Loki-File-Server-Verb" : "POST",
+                "X-Loki-File-Server-Headers" : "{ X-Loki-File-Server-Ephemeral-Key : \(keyPair.publicKey.base64EncodedString()) }",
+                "Connection" : "close"
+            ]
+            let (promise, resolver) = LokiAPI.RawResponsePromise.pending()
+            let proxyRequest = AFHTTPRequestSerializer().request(withMethod: "POST", urlString: url, parameters: nil, error: nil)
+            proxyRequest.allHTTPHeaderFields = proxyRequestHeaders
+            proxyRequest.httpBody = try JSONSerialization.data(withJSONObject: [ "cipherText64" : ivAndCipherText.base64EncodedString() ], options: [])
+            proxyRequest.timeoutInterval = request.timeoutInterval
+            var task: URLSessionDataTask!
+            task = httpSession.dataTask(with: proxyRequest as URLRequest) { response, result, error in
+                if let error = error {
+                    let nmError = NetworkManagerError.taskError(task: task, underlyingError: error)
+                    let nsError: NSError = nmError as NSError
+                    nsError.isRetryable = false
+                    resolver.reject(nsError)
+                } else {
+                    resolver.fulfill(result)
+                }
+            }
+            task.resume()
+            return promise
+        }.map { rawResponse in
+            guard let data = rawResponse as? Data, !data.isEmpty, let cipherText = Data(base64Encoded: data) else {
+                print("[Loki] Received a non-string encoded response.")
+                return rawResponse
+            }
+            let response = try DiffieHellman.decrypt(cipherText, using: symmetricKey)
+            let uncheckedJSON = try? JSONSerialization.jsonObject(with: response, options: .allowFragments) as? JSON
+            guard let json = uncheckedJSON, let httpStatusCode = json["status"] as? Int else { throw HTTPError.networkError(code: -1, response: nil, underlyingError: Error.proxyResponseParsingFailed) }
+            let isSuccess = (200..<300).contains(httpStatusCode)
+            var body: Any? = nil
+            if let bodyAsString = json["body"] as? String {
+                body = bodyAsString
+                if let bodyAsJSON = try? JSONSerialization.jsonObject(with: bodyAsString.data(using: .utf8)!, options: .allowFragments) as? [String: Any] {
+                    body = bodyAsJSON
```

Comment:
After doing this, you might want to rip out the status code from inside this body object too, since that will give you the true response code from the target.